src/osd/PrimaryLogPG.cc

```diff
@@ -10593,6 +10593,11 @@ hobject_t PrimaryLogPG::get_fpoid_from_chunk(const hobject_t soid, bufferlist& c
   pg_t raw_pg;
   object_locator_t oloc(soid);
   oloc.pool = pool.info.get_dedup_tier();
+  // check if dedup_tier isn't set
+  if (oloc.pool <= 0) {
+    dout(10) << __func__ << " failed to get  pool info of the chunk object" << dendl;
+    return hobject_t();
```

Comment:
The caller doesn't check for hobject_t() as a return value.  The assert here was fine, but start_dedup needs to check for the validity of oloc.pool and return an error that will actually be propagated to the client from the CEPH_OSD_OP_CACHE_FLUSH case.  We also need a test for that case to confirm that the correct error gets propagated.

Reply:
Thank you for further explanation. I misunderstood what you intended. 
Changed to check dedup_tier is set in start_dedup and put assert back in get_fpoid_from_chunk.